DMTN-148.tex

```diff
@@ -50,9 +50,255 @@
+
+\subsection{Validation}
+
+Calibration products are not currently validated, largely due to the
+opaque construction process.  Some set of proper validation tasks must
+be made, applying the newly constructed product to a set of test
+exposures prior to that calibration product's certification.  Applying
+these validation tasks continuously as new data is taken will inform
+the date ranges for which the product is certified.
+
+\subsection{Certification}
+
+Once a calibration product has been confirmed through validation to
+correct the instrument effect it is designed for, it will be certified
+with a date range denoting when the calibration is valid, and assigned
+to a particular calibration collection (which may contain calibrations
+the same OIDF values, as the validity range adds another dimension).
+For FITS image calibrations, this is equivalent to a database
+operation; no persisted data need change.  This is also acceptable for
+text calibrations, although conversion to FITS (using the
+CalibrationBaseClass methods) may be desired for IO speed. In
+addition, as the text calibrations are human readable and easier to
+manage, they should be exported to the appropriate \verb|obs_CAMERA_data|
+package.  This provides a common versioned repository for the state of
+the camera calibration that can be exported and used independently of
+the Gen3 repository where they were constructed.  These exports should
+also include the configuration and input data records.
+
+\subsection{Modifying Calibrations}
+
+Most calibration products need user intervention only to launch the
+construction and to validate and certify the result.  However, it is
+assumed that some of the products used (such as \verb|CAMERA| and
+\verb|QE_CURVE| definitions) will not have a PipelineTask construction
+process.  The version of these products stored in the
+\verb|obs_CAMERA_data| (or \verb|obs_CAMERA| for the case of the
+\verb|CAMERA| calibration) repository will be ingested into the Butler
+repository as needed.  These products should still have as much
+configuration and provenance data as is possible, but should also note
+any manual edits with a name and date.  Enforcing the inclusion of the
+\verb|obs_CAMERA_data| packages git repository SHA1 entry into the
+metadata of calibration products that have been ingested from those
+packages should provide sufficient information to reproduce or revert
+the manual edits.  This provenance method should also be used for
+calibrations that need to be augmented in non-algorithmic ways, such
+as \verb|DEFECTS|.
+
+\subsection{Deprecating Calibrations}
+
+Calibrations that should no longer be used (due to code improvements
+or newly discovered problems) can be deprecated in Gen3 with
+operations on the calibration collection that contains them.  This
+will require tools for collection management, but this will be
+entirely Butler database operations and will likely not be calibration
+specific.
```

Comment:
But if those products are also represented in git repositories, how do we prevent them becoming un-deprecated every time that repository is re-ingested?

Reply:
I don't imagine that we'll reingest the git repositories, other than when creating a new Gen3 registry.  Even if we did, using the SHA1 method mentioned in the DMTN, reingesting the git repo will do nothing, as it will be attempting to ingest the same files into the same Butler collection, which will create a violation of the collection rules against the same dataId pointing to multiple datasets.